### PR TITLE
Change log for missing rancher-compose file

### PR DESCRIPTION
**Pull request by mrajashree**
https://github.com/rancher/rancher/issues/6306

### CODE REVIEW

**Comment by mrajashree** [comment]
The` readRancherCompose()` function was incorrectly returning `nil` even when there was an error so that has been changed. If `readRancherCompose` returns error to `walkCatalog`, that template version should be skipped. So instead of showing the 'Skipping template' message as an error message it is changed to log.Debug message. A check has been added in `readFile` to see if the file is present, and if not it is logged and the function returns. So the error message of `Error reading file ` isn't repeated in the log

File: manager/catalog.go [hunk 1]
```diff
@@ -320,6 +320,12 @@ func readFile(relativePath string, fileName string) (*[]byte, error) {
 		log.Errorf("Error forming path to file %s, error: %v", relativePath+"/"+fileName, err)
 	}
 
+	_, err = os.Stat(filename)
+	if os.IsNotExist(err) {
+		log.Debugf("File %s does not exist", relativePath+"/"+fileName)
```
Comment (joshwget) [inline manager/catalog.go]: Can you use `path.Join` for this and save it to a variable so it doesn't have to be done multiple times?

File: manager/catalog.go [hunk 2]
```diff
@@ -174,7 +174,7 @@ func (cat *Catalog) walkCatalog(path string, f os.FileInfo, err error) error {
 						if ValidationMode {
 							log.Fatalf("Error processing the template version: %s, error: %v", f.Name()+"/"+subfile.Name(), err)
 						}
-						log.Errorf("Skipping the template version: %s, error: %v", f.Name()+"/"+subfile.Name(), err)
+						log.Debugf("Skipping the template version: %s, error: %v", f.Name()+"/"+subfile.Name(), err)
```
Comment (joshwget) [inline manager/catalog.go]: Same as other comment here

**Comment by mrajashree** [comment]
@joshwget did the changes, can you please review this again?

File: manager/catalog.go [hunk 3]
```diff
@@ -315,14 +317,21 @@ func readRancherCompose(relativePath string, newTemplate *model.Template) error
 }
 
 func readFile(relativePath string, fileName string) (*[]byte, error) {
-	filename, err := filepath.Abs(relativePath + "/" + fileName)
+	filePath := path.Join(relativePath, fileName)
+	filename, err := filepath.Abs(filePath)
 	if err != nil {
-		log.Errorf("Error forming path to file %s, error: %v", relativePath+"/"+fileName, err)
+		log.Errorf("Error forming path to file %s, error: %v", filePath, err)
```
Comment (joshwget) [inline manager/catalog.go]: Wouldn't we want to return the error here?

Comment (mrajashree) [inline manager/catalog.go]: @joshwget yes we should. This was already there in the code so I missed it while adding changes

**Comment by joshwget** [comment]
LGTM

**Pull request merged**